src/cript/nodes/util/__init__.py

```diff
@@ -38,6 +38,8 @@ class NodeEncoder(json.JSONEncoder):
         A set to store the node types that should be condensed to UUID edges in the JSON.
     suppress_attributes : Optional[Dict[str, Set[str]]]
         A dictionary that allows suppressing specific attributes for nodes with the corresponding UUIDs.
+    suppress_uid_usage : bool
+        Can suppress the usage of UID abbreviations. Warning, this can lead to infinite recursion with graphs that contain cycles.
```

Comment:
Idea: we could have two depth limits and a `depth` argument passed in recursive calls.

For example:
- `fullNodeDepth` would serialize to full nodes until this depth level is reached

Example:

F: full node
E: edge
fullNodeDepth: 4

F*: are the same node, serialized differently due to the backreference

```
F -> F -> F -> F* -> E
   \
     -> F -> F* ->F -> E
           \
              -> F -> F -> E
```